react:Hooks index page translation (#464)
reactjs/fr.react.dev: src/content/reference/react/index.md

---

> title: "Hooks intégrés à React"

Author: > Au passage ajuste aussi le titre correspondant dans `src/sidebarLearn.json` s'il te plaît

hello @tdd, merci beaucoup pour votre review. j'ai verifier dans le fichier en question mais le titre n'existe pas.

> title: "Hooks intégrés à React"

Reviewer: Ah oui on n'est pas dans Learn. C'est le fichier de sidebar pour la partie référence qu'il faut aller voir.

> title: "Hooks intégrés à React"

Author: j'ai vérifié, il ne s'y trouve non plus dans le fichier. je pense qu'il a été changé puisque le titre dans le ficher est `Hooks` tout court

  // ...
```

Les Effets sont une "issue de secours" du paradigme de React. N'utilisez pas les Effets pour orchestrer le flux de données de votre application. Si vous n'intéragissez pas avec un système externe, [Vous n'avez pas besoin d'un Effet.](/learn/you-might-not-need-an-effect)

Il existe deux variantes rarement utilisées de `useEffect` avec des différences de timing :

* [`useLayoutEffect`](/reference/react/useLayoutEffect) se déclenche avant que le navigateur ne repeigne l'écran. Vous pouvez mesurer la mise en page ici.
* [`useInsertionEffect`](/reference/react/useInsertionEffect) se déclenche avant que React ne fasse des changements dans le DOM. Les bibliothèques peuvent insérer du CSS dynamique ici.

---

## Les Hooks de performance {/*performance-hooks*/}

> La manière courante d'optimiser la performance de réaffichage est d'éviter les tâches inutiles. Par exemple, vous pouvez demander à React de réutiliser la valeur mise en cache ou d'éviter un réaffichage si la donnée n'a pas changé depuis l'affichage précedent.

Reviewer: ```suggestion
Une manière courante d’optimiser la performance de réaffichage consiste à s’épargner des tâches superflues. Par exemple, vous pouvez demander à React de réutiliser un calcul mis en cache ou d'éviter un nouveau rendu si la donnée n'a pas changé depuis le rendu précédent.
```

- Attention au mot-à-mot, le "est" entre deux propositions c'est inélégant. Ne pas hésiter à reformuler un peu.
- Le premier "réaffichage" est bien ("rerendu", blerk) car on est sur un verbe, ça reste concis. Le deuxième est un nom, "nouveau rendu" dans le contexte de la phrase sera mieux, et on utilise souvent "rendu" pour "render" en tant que nom dans les trads ici.

Author: > Le premier "réaffichage" est bien ("rerendu", blerk) car on est sur un verbe, ça reste concis. Le deuxième est un nom, "nouveau rendu" dans le contexte de la phrase sera mieux, et on utilise souvent "rendu" pour "render" en tant que nom dans les trads ici.

exactement j'avais d'énorme doute sur la VF de render/re-render, je ne savais pas quoi mettre. je pense que c'est encore mieux d'ajouter certains mots dans le glossaire. qu'en pensez vous 🙂 ?

Reviewer: J'en pense que le Glossaire a déjà une entrée de base + une section dédiée sur ces sujets 😉

> Pour éviter les calculs coûteux et les réaffichages inutiles, utilisez l'un de ces Hooks:

Reviewer: ```suggestion
Pour éviter les calculs coûteux et les réaffichages inutiles, utilisez l’un de ces Hooks :
```

(En pratique je fais toujours une repasse locale pour check visuel + typographie, mais autant avancer dès maintenant…)

- [`useMemo`](/reference/react/useMemo) vous permet de mettre en cache le résultat d'un calcul coûteux.

> - [`useCallback`](/reference/react/useCallback) Vous permet de mettre en cache la création d'une fonction avant de la passer à un composant optimisé.

Reviewer: ```suggestion
- [`useCallback`](/reference/react/useCallback) vous permet de mettre en cache la définition d’une fonction avant de la passer à un composant optimisé.
```

}
```

> Parfois, vous ne pouvez pas éviter le réaffichage parce que la vue actuelle doit être mis à jour. Dans ce cas, vous pouvez améliorer la performance en séparant les mises à jour bloquantes qui sont synchrones (comme si vous tapiez dans un champ de saisie) aux mises à jour non bloquantes qui ne bloquent pas l'interface utilisateur (mise à jour d'une carte).

Reviewer: ```suggestion
Parfois, vous ne pouvez pas éviter le réaffichage parce que la vue doit effectivement être mise à jour. Dans ce cas, vous pouvez améliorer la performance en séparant les mises à jour bloquantes qui doivent être synchrones (comme la gestion d'une saisie dans un champ) des mises à jour non bloquantes, qui ne gèlent  pas le traitement de l'interface (comme la mise à jour d'un graphique).
```

- Attention au contresens sur 'actually" ("véritablement") / actuel ("current")
- Chart = graphique (carte = map)
- Allègement des redondances
- Attention au glissement de sens (ce n'est pas forcément nous qui saisissons, etc.)

> Pour établir un ordre de priorité d'affichage, utilisez un de ces Hooks:

Reviewer: ```suggestion
Pour établir des priorités de rendu, utilisez un de ces Hooks :
```

> - [`useTransition`](/reference/react/useTransition) permet de marquer une transition d'état local comme non bloquante et d'autoriser d'autres mises à jour à l'interrompre.

Reviewer: ```suggestion
- [`useTransition`](/reference/react/useTransition) permet de marquer une transition d'état local comme non bloquante, ce qui autorise d'autres mises à jour à l'interrompre.
```

> - [`useDeferredValue`](/reference/react/useDeferredValue) vous permet de différer la mise à jour d'une partie non critique de l'interface utilisateur et de laisser les autres parties se mettre à jour en premier.

Reviewer: ```suggestion
- [`useDeferredValue`](/reference/react/useDeferredValue) vous permet de différer la mise à jour d'une partie non critique de l’UI et de laisser les autres parties se mettre à jour en premier.
```

---

## Autres Hooks {/*other-hooks*/}

> Ces Hooks sont majoritairement utiles aux auteurs de bibliothèque et ne sont pas couramment utilisés dans le code de l'application.

Reviewer: ```suggestion
Ces Hooks sont majoritairement utiles aux auteur·e·s de bibliothèque et ne sont pas couramment utilisés dans du code applicatif.
```

La VO pourrait se passer du "the" ici.

> - [`useDebugValue`](/reference/react/useDebugValue) vous permet de personnaliser le titre que React Devtools affiche pour votre Hook personnalisé.

Reviewer: ```suggestion
- [`useDebugValue`](/reference/react/useDebugValue) vous permet de personnaliser le libellé que React Devtools affiche pour votre Hook personnalisé.
```

> - [`useId`](/reference/react/useId) permet à un composant de s'associer lui-même à un ID unique. Géneralement avec les accessibilités des APIs.

Reviewer: ```suggestion
- [`useId`](/reference/react/useId) permet à un composant de s'associer un ID unique. Généralement utilisé avec les API d’accessibilité.
```

Contresens sur la fin.

> - [`useSyncExternalStore`](/reference/react/useSyncExternalStore) permet à un composant de s'abonner à une source de données externe.

Reviewer: ```suggestion
- [`useSyncExternalStore`](/reference/react/useSyncExternalStore) permet à un composant de s'abonner à une source de données extérieure.
```

---

## Vos propres Hooks {/*your-own-hooks*/}

> Vous pouvez [définir vos propres Hooks personnalisés](/learn/reusing-logic-with-custom-hooks#extracting-your-own-custom-hook-from-a-component) comme des fonctions Javascript.

Reviewer: ```suggestion
Vous pouvez [définir vos propres Hooks personnalisés](/learn/reusing-logic-with-custom-hooks#extracting-your-own-custom-hook-from-a-component) au moyen de fonctions JavaScript.
```
